addons/web/static/src/core/macro.js

```diff
@@ -126,34 +162,62 @@ class Macro {
 }
 
 export class MacroEngine {
-    constructor(target = document.body) {
+    constructor(params) {
+        const { target, defaultCheckDelay } = Object.assign(
+            {
+                target: document.body,
+                defaultCheckDelay: 750,
+            },
+            params
+        );
         this.isRunning = false;
         this.timeout = null;
         this.target = target;
-        this.interval = Infinity; // nbr of ms before we check the dom to advance macros
+        this.defaultCheckDelay = defaultCheckDelay;
         this.macros = new Set();
+        this.observerOptions = {
+            attributes: true,
+            childList: true,
+            subtree: true,
+            characterData: true,
+        };
         this.observer = new MutationObserver(this.delayedCheck.bind(this));
+        this.iframeObserver = new MutationObserver(() => {
+            const iframeEl = document.querySelector("iframe.o_iframe");
```

Comment:
this seems sad to perform a queryselector each time. is there a reason not to check the mutation events received by the observer?

Reply:
I checked website which is uses this feature. The iframe used doesn't seem to be added since it can't be found from the "mutation.addedNodes". Perhaps the iframe is server-rendered.